### PR TITLE
feat(popover): enable closing on outside click

### PR DESCRIPTION
**Pull request by jjsino**
This PR will handle the outside click when popover tooltip is opened and close the popover

### CODE REVIEW

**Comment by rvanbaalen** [comment]
Please squash your commits.

File: src/popover/test/popover.spec.js [hunk 1]
```diff
@@ -47,6 +47,16 @@ describe('popover', function() {
     expect( tooltipScope.isOpen ).toBe( false );
   }));
 
+  it('should close on outside click', inject(function($document){
```
Comment (wesleycho) [inline src/popover/test/popover.spec.js]: Add space before `{`

File: src/tooltip/tooltip.js [hunk 2]
```diff
@@ -330,11 +339,18 @@ angular.module( 'ui.bootstrap.tooltip', [ 'ui.bootstrap.position', 'ui.bootstrap
 
               triggers = getTriggers( val );
 
-              if ( triggers.show === triggers.hide ) {
-                element.bind( triggers.show, toggleTooltipBind );
-              } else {
-                element.bind( triggers.show, showTooltipBind );
-                element.bind( triggers.hide, hideTooltipBind );
+              if (triggers.show != 'manual') {
```
Comment (wesleycho) [inline src/tooltip/tooltip.js]: Use `!==`

File: src/tooltip/tooltip.js [hunk 3]
```diff
@@ -330,11 +339,18 @@ angular.module( 'ui.bootstrap.tooltip', [ 'ui.bootstrap.position', 'ui.bootstrap
...
+                if (triggers.show === 'popover') {
+                  triggers.show = 'click';
+                  element.bind(triggers.show, toggleTooltipBind);
+                  $document.find('body').bind(triggers.show, hidePopoverBind);
+                } else if (triggers.show === triggers.hide) {
+                  element.bind(triggers.show, toggleTooltipBind);
+                }
+                else {
```
Comment (wesleycho) [inline src/tooltip/tooltip.js]: Change to `} else {`

**Comment by wesleycho** [comment]
This needs documentation.

**Comment by jjsino** [comment]
Squash commits done. Please check it out.

File: src/tooltip/tooltip.js [hunk 4]
```diff
@@ -330,11 +339,17 @@ angular.module( 'ui.bootstrap.tooltip', [ 'ui.bootstrap.position', 'ui.bootstrap
 
               triggers = getTriggers( val );
 
-              if ( triggers.show === triggers.hide ) {
-                element.bind( triggers.show, toggleTooltipBind );
-              } else {
-                element.bind( triggers.show, showTooltipBind );
-                element.bind( triggers.hide, hideTooltipBind );
+              if (triggers.show !== 'manual') {
```
Comment (wesleycho) [inline src/tooltip/tooltip.js]: Why is this necessary? As far as I can tell, there is no reference to `manual` at all - this if conditional should be removed.

File: src/tooltip/tooltip.js [hunk 5]
```diff
@@ -330,11 +339,17 @@ angular.module( 'ui.bootstrap.tooltip', [ 'ui.bootstrap.position', 'ui.bootstrap
...
+                if (triggers.show === 'popover') {
```
Comment (wesleycho) [inline src/tooltip/tooltip.js]: Can you explain why this needs to be handled special?

File: src/popover/test/popover.spec.js [hunk 6]
```diff
@@ -47,6 +47,16 @@ describe('popover', function() {
     expect( tooltipScope.isOpen ).toBe( false );
   }));
 
+  it('should close on outside click', inject(function($document) {
```
Comment (wesleycho) [inline src/popover/test/popover.spec.js]: Remove `inject` and `$document`

File: src/tooltip/tooltip.js [hunk 7]
```diff
@@ -177,6 +179,12 @@ angular.module( 'ui.bootstrap.tooltip', [ 'ui.bootstrap.position', 'ui.bootstrap
               });
             }
 
+            function hidePopoverBind(e) {
+              if (!element.parent()[0].contains(e.target)) {
```
Comment (wesleycho) [inline src/tooltip/tooltip.js]: This needs to be tested in IE

**Comment by wesleycho** [comment]
Can you rebase and address the comments?

**Comment by wesleycho** [comment]
Closing this as superceded by the work in 873730360c84baeb81957704f991deed495358c8

**Pull request closed without merging**